src/main/java/hudson/plugins/sonar/SonarBuildWrapper.java

```diff
@@ -71,56 +70,52 @@ public void setUp(Context context, Run<?, ?> build, FilePath workspace, Launcher
     Logger.LOG.info(msg);
     listener.getLogger().println(msg);
 
-    context.getEnv().putAll(createVars(installation));
+    context.getEnv().putAll(createVars(installation, initialEnvironment));
 
     context.setDisposer(new AddBuildInfo(installation));
 
     build.addAction(new SonarMarkerAction());
   }
 
   @VisibleForTesting
-  static Map<String, String> createVars(SonarInstallation inst) {
+  static Map<String, String> createVars(SonarInstallation inst, EnvVars initialEnvironment) {
     Map<String, String> map = new HashMap<>();
 
     map.put("SONAR_CONFIG_NAME", inst.getName());
-    String hostUrl = getOrDefault(inst.getServerUrl(), "http://localhost:9000");
+    String hostUrl = getOrDefault(initialEnvironment.expand(inst.getServerUrl()), "http://localhost:9000");
     map.put("SONAR_HOST_URL", hostUrl);
-    String login = getOrDefault(inst.getSonarLogin(), "");
-    map.put("SONAR_LOGIN", login);
-    String password = getOrDefault(inst.getSonarPassword(), "");
-    map.put("SONAR_PASSWORD", password);
-    String token = getOrDefault(inst.getServerAuthenticationToken(), "");
+    String token = getOrDefault(initialEnvironment.expand(inst.getServerAuthenticationToken()), "");
     map.put("SONAR_AUTH_TOKEN", token);
 
-    map.put("SONAR_JDBC_URL", getOrDefault(inst.getDatabaseUrl(), ""));
-
-    String jdbcDefault = SQServerVersions.SQ_5_1_OR_LOWER.equals(inst.getServerVersion()) ? DEFAULT_SONAR : "";
-    map.put("SONAR_JDBC_USERNAME", getOrDefault(inst.getDatabaseLogin(), jdbcDefault));
-    map.put("SONAR_JDBC_PASSWORD", getOrDefault(inst.getDatabasePassword(), jdbcDefault));
-
     if (StringUtils.isEmpty(inst.getMojoVersion())) {
       map.put("SONAR_MAVEN_GOAL", "sonar:sonar");
     } else {
       map.put("SONAR_MAVEN_GOAL", SonarUtils.getMavenGoal(inst.getMojoVersion()));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Annotate the parameter with @javax.annotation.Nullable in method 'getMavenGoal' declaration, or make sure that null can not be passed as argument. ([squid:S4449](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS4449))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.jenkins-ci.plugins%3Asonar&pullRequest=90&issues=AWI-fL1adpSr17abUILf&open=AWI-fL1adpSr17abUILf)

[ISSUE_KEY]: AWI-fL1adpSr17abUILf